### ISSUE TITLE
Segfault on deleting remote directory

### ISSUE THREAD

**Issue by nighti** [skilion/onedrive#134]
I have delete a local onedrive folder, after restart the process segfaults.
Host is a x86_64 Fedora with onedrive compiled yesterdays version.

```
Deleting remote item: ./Bilder/Camera Roll
sync.SyncException@src/sync.d(348): HTTP request returned status code 412 (Precondition Failed)
{
    "error": {
        "code": "notAllowed",
        "innererror": {
            "code": "preconditionFailed"
        },
        "message": "ETag does not match current item's value"
    }
}
----------------
??:? void sync.SyncEngine.scanForDifferences(immutable(char)[]) [0x5c62a2]
??:? void main.performSync(sync.SyncEngine) [0x5b43e6]
??:? _Dmain [0x5b3c02]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5dacfe]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5dac48]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5dacba]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5dac48]
??:? _d_run_main [0x5dabb9]
??:? main [0x5b6c85]
??:? __libc_start_main [0x2e658400]
onedrive.OneDriveException@src/onedrive.d(380): HTTP request returned status code 412 (Precondition Failed)
{
    "error": {
        "code": "notAllowed",
        "innererror": {
            "code": "preconditionFailed"
        },
        "message": "ETag does not match current item's value"
    }
}
----------------
??:? void onedrive.OneDriveApi.del(const(char)[]) [0x5bbae8]
??:? void sync.SyncEngine.uploadDeleteItem(itemdb.Item, const(char)[]) [0x5c9236]
??:? void sync.SyncEngine.uploadDifferences(itemdb.Item) [0x5c67ab]
??:? void sync.SyncEngine.uploadDifferences(itemdb.Item) [0x5c671c]
??:? void sync.SyncEngine.uploadDifferences(itemdb.Item) [0x5c671c]
??:? void sync.SyncEngine.scanForDifferences(immutable(char)[]) [0x5c6139]
??:? void main.performSync(sync.SyncEngine) [0x5b43e6]
??:? _Dmain [0x5b3c02]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5dacfe]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5dac48]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5dacba]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5dac48]
??:? _d_run_main [0x5dabb9]
??:? main [0x5b6c85]
??:? __libc_start_main [0x2e658400]
Segmentation fault (Speicherabzug geschrieben)
```

**Comment by skilion** [comment]
This bug has been recently fixed in 0afb6fa3923066f46c531f33e175011cd69e06a8, try to update you repository

**Comment by nighti** [comment]
With latest update fixed.
Thx!!
